### ISSUE TITLE
[BUG] Failed to create image when deployed in private network environment

### ISSUE THREAD

**Issue by johnliu55tw** [harvester/harvester#1012]
**Describe the bug**
I've deployed Harvester using ISO in a private network, where HTTP Proxy is required to fetch external URLs. During the installation, the HTTP Proxy was set accordingly.

I tried to create an image using an external URL (https://cloud-images.ubuntu.com/focal/20210707/focal-server-cloudimg-amd64.img), it failed and shows error message:
> Head https://cloud-images.ubuntu.com/focal/20210707/focal-server-cloudimg-amd64.img: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
![image](https://user-images.githubusercontent.com/5047724/125034010-a915fd80-e0c2-11eb-8d73-5a6469d6a6ae.png)

Then I SSHed into harvester and run command `sudo k3os config --dump` then I notice there was no `http_proxy` and `https_proxy` inside the config.

To check if it's other networking issues, I set up an Nginx HTTP server to serve the same image within the private network, then Harvester can create the image without any issue.
![image](https://user-images.githubusercontent.com/5047724/125035462-6bb26f80-e0c4-11eb-9392-b2ebb12e33c3.png)

**To Reproduce**

1. Install Harvester using ISO, and properly set up HTTP Proxy information.
2. Wait until Harvester to enter ready state.
3. Create an image using URL https://cloud-images.ubuntu.com/focal/20210707/focal-server-cloudimg-amd64.img
4. The creation **failed** with error message: `Head https://cloud-images.ubuntu.com/focal/20210707/focal-server-cloudimg-amd64.img: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`

**Expected behavior**
Create images with external URLs in a private network using HTTP Proxy should be fine.

**Support bundle**

**Environment:**
 - Harvester ISO version: harvester-v0.2.0-amd64.iso
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Baremetal

**Additional context**
I've tried updated HTTP Proxy information by adding `/var/lib/rancher/k3os/config.yaml` with following content:
```
k3os:
  environment:
    http_proxy: http://our-proxy:8080/
    https_proxy: http://our-proxy:8080/
```
Then enter the command `sudo k3os config` but this doesn't seem to work. I'm not familiar with k3os, maybe there's a way to configure HTTP Proxy manually?

**Comment by gitlawr** [comment]
The proxy configuration of the installer is used to fetch remote config files during the installation. It takes no effect afterward.

Configuring in the OS environment does no help because jobs are done in containers.
Theoretically, it works if we set http_proxy envs to harvester server pods and longhorn backing-image-manager pods(who do the download work) but it seems that backing-image-manager pods are not configurable wrt envs. We might need to investigate how to make the proxy configuration easier, e.g. have a mutating webhook to inject proxy envs. 

In short, the workaround in the current release is to serve images or use a proxy as you did in the private environment.

**Comment by TachunLin** [comment]
@guangbochen @johnliu55tw 
 
Verified **not fixed** on master-d05a6cc7-head. Move issue to review stage. 

### Result
Use iso deploy harvester in private network (isolated) and configure http proxy during installation 
Can't create an image using an external URL, it still show error message
`dial tcp: lookup cloud-images.ubuntu.com on 10.53.0.10:53: server misbehaving`


![image](https://user-images.githubusercontent.com/29251855/138991104-fbea2c2c-6193-43d9-aa31-7a6f2afe19ed.png)

![image](https://user-images.githubusercontent.com/29251855/138991256-c0ffcc58-3971-430e-9b1c-ec1d1b0def86.png)


### Test Information
* Test Environment: 1 local qemu kvm machine for harvester nodes 
                                1 local qemu kvm machine for http proxy (debian 11, with tinyproxy)
                                1 isolated virtual network add to harvester and http proxy server 
                                1 NAT virtual network add to http proxy server 

![image](https://user-images.githubusercontent.com/29251855/138991715-b1ad6c9b-894d-41c0-8e2f-7871f2101586.png)

                                
* Harvester version: master-d05a6cc7-head

### Environment Preparation

1. Create a isolated virtual network
2. Create a debian vm, add isolated and NAT network
3. Setup and configure tinyproxy on debian vm for http proxy server
4. Create new vm and ISO install harvester
5. Set static ip address in isolated network and gateway for node
6. Set static dns server 
7. Use DHCP to retrieve VIP
8. Provide http proxy ( ex. http://192.168.100.135:8888)
9. Harvester in ready status and can login to dashboard

### Verify Steps

1. Environment preparation as above steps
2. Login harvester dashboard
3. Open images page
4. Create an image by using URL

### Additional Context 
Test export http_proxy connection failed in virtual terminal  
![image](https://user-images.githubusercontent.com/29251855/138993078-3f9a8697-b743-4372-8de7-b9ac50815483.png)

**Comment by guangbochen** [comment]
my bad, the PR is not merged, should be a mistake of moving the issue around. Will be supported via https://github.com/harvester/harvester/pull/1337

**Comment by gitlawr** [comment]
Awaiting for UI changes

**Comment by TachunLin** [comment]
Verified **fixed** on master-45e25939-head. Close this issue. 

### Result
With the new http proxy feature provide in #1218 , we can set proxy value in harvester settings to create image from URL through the internal proxy server
![image](https://user-images.githubusercontent.com/29251855/142406004-b524e348-79a9-48a2-b667-52c44b22dc77.png)

![image](https://user-images.githubusercontent.com/29251855/142406060-534c3c53-7e7e-42e1-ae22-60de4cb3fee6.png)

### Test Information
* Test Environment: 3 harvester nodes on local kvm machines
* Network: airgapped network created by vagrant pxe example
* Http proxy: squid server on pxe_server                           
* Harvester version: master-45e25939-head

### Verify Steps

1. Clone the ipxe example in the offline feature branch
2. Create 3 nodes harvester by using vagrant pxe example
3. Open settings
4. Provide the following value in the http-proxy settings
```
{
  "httpProxy": "http://192.168.0.254:3128",
  "httpsProxy": "http://192.168.0.254:3128",
  "noProxy": "localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,cattle-system.svc,192.168.0.0/16,.svc,.cluster.local,example.com"
}
```
5. Create image from URL 
https://cloud-images.ubuntu.com/focal/20211116/focal-server-cloudimg-amd64.img
